COMMIT_GUIDELINES.md

```diff
@@ -0,0 +1,149 @@
+# General philosophy
+
+- **Each change goes into its own commit**. If you want to summarize what you did with this commit (in the commit message)
+and you start using the word "and", you probably want to split the commit up into 2 or more individual commits.
+
+- **Each commit should compile** (try to make each commit self-contained so that it is possible to compile the code at
+each commit)
```

Comment:
```suggestion
- **Each commit should compile** (try to make each commit self-contained so that it is possible to compile the code at
each commit).
```

Reply:
Done